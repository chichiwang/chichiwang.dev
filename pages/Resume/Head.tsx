/* global PROJECT_URL */
import React from 'react';

function Head() {
  const url = `${PROJECT_URL}`;

  return (
    <>
      <meta property="og:type" content="website" />
      <meta property="og:title" content="Resume of Chi-chi Wang" />
      <meta property="og:description" content="Software engineer, UI specialist, JavaScript expert, team leader, mentor." />
      <meta property="og:url" content={url} />
      <link href={url} rel="canonical" />
    </>
  );
}

export default Head;
