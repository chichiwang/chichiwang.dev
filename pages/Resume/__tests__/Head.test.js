import React from 'react';
import renderer from 'react-test-renderer';

import Head from '../Head';

describe('Resume/Head component', () => {
  beforeAll(() => {
    global.PROJECT_URL = 'https://resume.head.test';
  });

  afterAll(() => {
    delete global.PROJECT_URL;
  });

  test('matches the snapshot', () => {
    const tree = renderer.create(<Head />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
