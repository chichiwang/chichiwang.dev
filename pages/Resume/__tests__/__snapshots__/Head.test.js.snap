// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume/Head component matches the snapshot 1`] = `
[
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="Resume of Chi-chi Wang"
    property="og:title"
  />,
  <meta
    content="Software engineer, UI specialist, JavaScript expert, team leader, mentor."
    property="og:description"
  />,
  <meta
    content="https://resume.head.test"
    property="og:url"
  />,
  <link
    href="https://resume.head.test"
    rel="canonical"
  />,
]
`;
